cluster/gce/gci/scaleoutserver-helper.sh

```diff
@@ -110,3 +110,90 @@ function create-proxy-instance-internal() {
   return 1
 }
 
+function create-scaleoutserver-instance {
+  local name=""
+  local address=""
+  local private_network_ip=""
+  [[ -n ${1:-} ]] && name="${1}"
+  [[ -n ${2:-} ]] && address="${2}"
+  [[ -n ${3:-} ]] && private_network_ip="${3}"
+  write-master-env
+  ensure-gci-metadata-files
+  create-scaleoutserver-instance-internal "${name}" "${address}" "${private_network_ip}"
+}
+
+function create-scaleoutserver-instance-internal() {
+  local gcloud="gcloud"
+  local retries=5
+  local sleep_sec=10
+  if [[ "${MASTER_SIZE##*-}" -ge 64 ]]; then  # remove everything up to last dash (inclusive)
+    # Workaround for #55777
+    retries=30
+    sleep_sec=60
+  fi
+
+  local -r server_name="${1}"
+  local -r address="${2:-}"
+  local -r private_network_ip="${3:-}"
+
+  local enable_ip_aliases
+  if [[ "${NODE_IPAM_MODE:-}" == "CloudAllocator" ]]; then
+    enable_ip_aliases=true
+  else
+    enable_ip_aliases=false
+  fi
+
+  local network=$(make-gcloud-network-argument \
+    "${NETWORK_PROJECT}" "${REGION}" "${NETWORK}" "${SUBNETWORK:-}" \
+    "${address:-}" "${private_network_ip:-}" "${enable_ip_aliases:-}" "${IP_ALIAS_SIZE:-}")
+
+  local metadata="kube-env=${KUBE_TEMP}/master-kube-env.yaml"
```

Comment:
how to ensure those files are instance specific ? is the ${kube_temp} recreated for each instance ?

Reply:
${kube_temp} created only one time when start Kube-up.

This code overwrites the file if it exists or creates it if it doesn't exist for each instance:
```
cat >$file <<EOF
```